### PR TITLE
[DCOS-50377] Fix readiness check for Kerberized Hdfs

### PR DESCRIPTION
**Pull request by ishanchandrajoshi**
This branch contains the code to upgrade Kerberized Hdfs.

### CODE REVIEW

File: frameworks/hdfs/keytab-fix/src/test/java/org/sample/AppTest.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    /**
+     * @return the suite of tests being tested
+     */
+    public static Test suite()
+    {
+        return new TestSuite( AppTest.class );
+    }
+
+    /**
+     * Rigourous Test :-)
```
Comment (samvantran) [inline frameworks/hdfs/keytab-fix/src/test/java/org/sample/AppTest.java]: ?? 

Please delete this file if there is no purpose for it.

File: frameworks/hdfs/keytab-fix/src/main/java/org/sample/KrbFixMain.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
+package org.sample;
```
Comment (samvantran) [inline frameworks/hdfs/keytab-fix/src/main/java/org/sample/KrbFixMain.java]: `org.sample`? Please use something more appropriate.

File: frameworks/hdfs/src/main/dist/svc.yml [hunk 3]
```diff
@@ -243,20 +243,20 @@ pods:
...
             {{/TASKCFG_ALL_SECURITY_KERBEROS_DEBUG}}
             KRB5_CONFIG=$MESOS_SANDBOX/{{TASKCFG_ALL_HDFS_VERSION}}/etc/hadoop/krb5.conf kinit -k -t $MESOS_SANDBOX/hdfs.keytab $SECURITY_KERBEROS_PRIMARY/$TASK_NAME.$FRAMEWORK_HOST@$SECURITY_KERBEROS_REALM
             {{/SECURITY_KERBEROS_ENABLED}}
+            export service_state=$(./{{TASKCFG_ALL_HDFS_VERSION}}/bin/hdfs haadmin -getServiceState $TASK_NAME | grep -i 'standby')
+            export LATEST_FSIMAGE_VERSION=-65
+            export CURRENT_FSIMAGE_VERSION=$(grep "layoutVersion" name-data/current/VERSION | cut -d'=' -f2)
+            if [[ $service_state -eq 'standby' && $FINALIZED -eq false && $POD_INSTANCE_INDEX -eq 1 && $CURRENT_FSIMAGE_VERSION -gt $LATEST_FSIMAGE_VERSION ]]; then
+              ./{{TASKCFG_ALL_HDFS_VERSION}}/bin/hdfs dfsadmin -rollingUpgrade finalize
+              FINALIZED=true
+            fi
```
Comment (samvantran) [inline frameworks/hdfs/src/main/dist/svc.yml]: Ah, all of the kerberos configs needed to be set first. Good find.

File: frameworks/hdfs/keytab-fix/src/main/java/org/sample/KrbTestMain.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
+package org.sample;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.regex.Pattern;
+
+public class KrbTestMain {
```
Comment (samvantran) [inline frameworks/hdfs/keytab-fix/src/main/java/org/sample/KrbTestMain.java]: Doesn't seem like this file is used anywhere

File: frameworks/hdfs/keytab-fix/pom.xml [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+      <groupId>junit</groupId>
+      <artifactId>junit</artifactId>
+      <version>3.8.1</version>
+      <scope>test</scope>
+    </dependency>
+    <!--<dependency>
+      <groupId>org.apache.hadoop</groupId>
+      <artifactId>hadoop-common</artifactId>
+      <version>3.2.0</version>
+    </dependency> -->
```
Comment (samvantran) [inline frameworks/hdfs/keytab-fix/pom.xml]: Please delete

File: frameworks/hdfs/src/main/dist/svc.yml (hunk 3 above)
Comment (ishanchandrajoshi) [reply frameworks/hdfs/src/main/dist/svc.yml]: Yes

**Comment by samvantran** [comment]
The title of this PR still reads `[DCOS-50377] Soak Hdfs 2.7.0-3.2.0 distribution`. Can you change it to something like 
```
[DCOS-50377] Fix readiness check for Kerberized HDFS
```

**Pull request merged**